fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/impl/attachment/sync/AttachmentSyncImpl.java

```diff
@@ -0,0 +1,171 @@
+	public static final ThreadLocal<Set<Identifier>> CLIENT_SUPPORTED_ATTACHMENTS = ThreadLocal.withInitial(Set::of);
+
+	public static AcceptedAttachmentsPayloadC2S createResponsePayload() {
+		return new AcceptedAttachmentsPayloadC2S(AttachmentRegistryImpl.getRegisteredAttachments());
+	}
+
+	public static void trySync(AttachmentSyncPayload payload, ServerPlayerEntity player) {
+		if (ServerPlayNetworking.canSend(player, PACKET_ID)) {
+			ServerPlayNetworking.send(player, payload);
+		}
+	}
+
+	private static Set<Identifier> decodeResponsePayload(AcceptedAttachmentsPayloadC2S payload) {
+		Set<Identifier> atts = payload.acceptedAttachments();
+		atts.retainAll(AttachmentRegistryImpl.getRegisteredAttachments());
+		return atts;
+	}
+
+	public static void refreshAttachments(ServerPlayerEntity player) {
+		if (ServerPlayNetworking.canSend(player, REFRESH_PACKET_ID)) {
+			ServerPlayNetworking.send(player, new AttachmentRefreshPayloadS2C(Optional.empty()));
+		}
+	}
+
+	@Override
+	public void onInitialize() {
+		// Config
+		PayloadTypeRegistry.configurationC2S()
+				.register(AcceptedAttachmentsPayloadC2S.ID, AcceptedAttachmentsPayloadC2S.CODEC);
+		PayloadTypeRegistry.configurationS2C()
+				.register(AcceptedAttachmentsPayloadS2C.ID, AcceptedAttachmentsPayloadS2C.CODEC);
+
+		ServerConfigurationConnectionEvents.CONFIGURE.register((handler, server) -> {
+			if (ServerConfigurationNetworking.canSend(handler, CONFIG_PACKET_ID)) {
+				handler.addTask(new AttachmentSyncTask());
+			}
+		});
+
+		ServerConfigurationNetworking.registerGlobalReceiver(AcceptedAttachmentsPayloadC2S.ID, (payload, context) -> {
+			// copied over from custom ingredient sync
+			Set<Identifier> supportedAttachments = decodeResponsePayload(payload);
+			ChannelHandler packetEncoder = ((ClientConnectionAccessor) ((ServerCommonNetworkHandlerAccessor) context.networkHandler()).getConnection())
+					.getChannel()
+					.pipeline()
+					.get("encoder");
+
+			if (packetEncoder != null) { // Null in singleplayer
+				((SupportedAttachmentsPacketEncoder) packetEncoder).fabric_setSupportedAttachments(supportedAttachments);
+			}
+
+			context.networkHandler().completeTask(AttachmentSyncTask.KEY);
+		});
+
+		// Play
+		PayloadTypeRegistry.playS2C().register(AttachmentSyncPayload.ID, AttachmentSyncPayload.CODEC);
+
+		PayloadTypeRegistry.playS2C().register(AttachmentRefreshPayloadS2C.ID, AttachmentRefreshPayloadS2C.CODEC);
+		PayloadTypeRegistry.playC2S().register(AttachmentRefreshPayloadC2S.ID, AttachmentRefreshPayloadC2S.CODEC);
+
+		ServerPlayNetworking.registerGlobalReceiver(AttachmentRefreshPayloadC2S.ID, (payload, context) -> {
+			List<AttachmentChange> refreshed = new ArrayList<>();
+
+			for (Map.Entry<AttachmentType<?>, AttachmentTargetInfo<?>> attachment : payload.attachments()) {
+				AttachmentTypeImpl<?> type = (AttachmentTypeImpl<?>) attachment.getKey();
+				AttachmentTarget target = attachment.getValue().getTarget(context.player().getServerWorld());
```

Comment:
I think this is deadly for a server, it allows a client to force load any chunk on the server. When doing any client -> server packets you really need to think them though carefully, as a client can send utter junk.